test/controllers/offer.controller.test.js

```diff
@@ -1931,6 +2002,214 @@ describe('Offer Controller', () => {
+              expect(sendMailStub).to.have.be.calledWith(EMAIL_CONTENT.OFFER_CREATED);
+              expect(sendMailStub.callCount).to.eq(1);
+              done();
+            });
+        });
+      });
+
+      context('when offer id is invalid', () => {
+        it('returns not found', (done) => {
+          request()
+            [method](`/api/v1/offer/${mongoose.Types.ObjectId()}/approve-terms`)
+            .set('authorization', adminToken)
+            .end((err, res) => {
+              expect(res).to.have.status(404);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Invalid offer');
+              expect(sendMailStub.callCount).to.eq(0);
+              done();
+            });
+        });
+      });
+
+      context('when offer does not require approval', () => {
+        beforeEach(async () => {
+          await Offer.findByIdAndUpdate(offer._id, { status: OFFER_STATUS.GENERATED });
+        });
+        it('returns error', (done) => {
+          request()
+            [method](endpoint)
+            .set('authorization', adminToken)
+            .end((err, res) => {
+              expect(res).to.have.status(412);
+              expect(res.body.success).to.be.eql(false);
+              expect(res.body.message).to.be.eql('Offer does not require approval');
+              expect(sendMailStub.callCount).to.eq(0);
+              done();
+            });
+        });
+      });
+
+      [vendorUser, regularUser].map((user) =>
+        itReturnsForbiddenForTokenWithInvalidAccess({
+          endpoint,
+          method,
+          user,
+          useExistingUser: true,
+        }),
+      );
+
+      itReturnsForbiddenForNoToken({ endpoint, method });
+
+      context('when approveTerms service fails', () => {
+        it('returns the error', (done) => {
+          sinon.stub(Offer, 'findOneAndUpdate').throws(new Error('Type Error'));
+          request()
+            [method](endpoint)
+            .set('authorization', adminToken)
+            .end((err, res) => {
+              expect(res).to.have.status(400);
+              expect(sendMailStub.callCount).to.eq(0);
+              done();
+              Offer.findOneAndUpdate.restore();
+            });
+        });
+      });
+    });
+
+    describe('Request Review for Terms & Conditions', () => {
+      const enquiry = EnquiryFactory.build(
+        { userId: regularUser._id, propertyId: properties[0]._id },
+        { generateId: true },
+      );
+      const offer = OfferFactory.build(
+        {
+          enquiryId: enquiry._id,
+          vendorId: vendorUser._id,
+          additionalClause: ['Additional Clause 1', 'Additional Clause 2'],
+          status: OFFER_STATUS.PENDING_ADMIN_APPROVAL,
+        },
+        { generateId: true },
+      );
+      const method = 'put';
+      const endpoint = `/api/v1/offer/${offer._id}/request-terms-review`;
+
+      const data = { comment: 'Please review clause 2' };
```

Comment:
```suggestion
      const data = { comment: 'Please review clause 2' };
```

I noticed that you used `comment` in the test for this instead of `question` used in the model. Lets update the model to `comment` for admin's comment so that it would be in sync. 

Please note that I think `comment` is a better representation in the model than `question` since the admin might not be asking a question.